[CI] GenerativeIT test failing

**Build Scans:**
- [elasticsearch-periodic #9209 / release-tests](https://gradle-enterprise.elastic.co/s/iq43wezutdzlg)
- [elasticsearch-pull-request #79259 / part-3](https://gradle-enterprise.elastic.co/s/rbsqjgs4dfoug)
- [elasticsearch-periodic #9129 / release-tests](https://gradle-enterprise.elastic.co/s/yv2vuqx4s7s72)
- [elasticsearch-pull-request #78956 / part-3](https://gradle-enterprise.elastic.co/s/mvgmosij6wrdq)
- [elasticsearch-periodic #9122 / release-tests](https://gradle-enterprise.elastic.co/s/jg44xh65oa2yq)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:multi-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT.test" -Dtests.seed=B920A31D98252F1C -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=sg-Latn-CF -Dtests.timezone=Asia/Amman -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: query: from col* | drop `primary`, `hex_code` | sort color NULLS LAST | rename rgb_vector AS `nKOaKCuWrfPT`
exception: method [POST], host [http://[::1]:33579], URI [/_query?format=yaml&pretty=true&error_trace=true], status line [HTTP/1.1 400 Bad Request]
Warnings: [No limit defined, adding default limit of [1000]]
---
error:
  root_cause:
  - type: "verification_exception"
    reason: "Found 1 problem\nline 1:73: Cannot use field [rgb_vector] with unsupported\
      \ type [dense_vector]"
    stack_trace: "org.elasticsearch.xpack.esql.VerificationException: Found 1 problem\n\
      line 1:73: Cannot use field [rgb_vector] with unsupported type [dense_vector]\n\
      \tat org.elasticsearch.xpack.esql.analysis.Analyzer.verify(Analyzer.java:205)\n\
      \tat org.elasticsearch.xpack.esql.analysis.Analyzer.analyze(Analyzer.java:199)\n\
      \tat org.elasticsearch.xpack.esql.session.EsqlSession.lambda$analyzedPlan$7(EsqlSession.java:350)\n\
      \tat org.elasti
[truncated]
```

**Issue Reasons:**
- [main] 3 consecutive failures in step release-tests
- [main] 5 failures in test test (3.7% fail rate in 136 executions)
- [main] 3 failures in step release-tests (100.0% fail rate in 3 executions)
- [main] 2 failures in step part-3 (3.6% fail rate in 55 executions)
- [main] 3 failures in pipeline elasticsearch-periodic (100.0% fail rate in 3 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (3.6% fail rate in 55 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -539,18 +539,12 @@ tests:
 - class: org.elasticsearch.multiproject.test.CoreWithMultipleProjectsClientYamlTestSuiteIT
   method: test {yaml=indices.resolve_index/10_basic_resolve_index/Resolve index with hidden and closed indices}
   issue: https://github.com/elastic/elasticsearch/issues/130568
-- class: org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT
-  method: test
-  issue: https://github.com/elastic/elasticsearch/issues/130067
 - class: org.elasticsearch.xpack.monitoring.exporter.http.HttpExporterTests
   method: testExporterWithHostOnly
   issue: https://github.com/elastic/elasticsearch/issues/130599
 - class: org.elasticsearch.xpack.monitoring.exporter.http.HttpExporterTests
   method: testCreateRestClient
   issue: https://github.com/elastic/elasticsearch/issues/130600
-- class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT
-  method: test
-  issue: https://github.com/elastic/elasticsearch/issues/130067
 - class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
   method: test {p0=search.vectors/40_knn_search/Dimensions are dynamically set}
   issue: https://github.com/elastic/elasticsearch/issues/130626

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/EsqlQueryGenerator.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/EsqlQueryGenerator.java
@@ -107,6 +107,10 @@ public class EsqlQueryGenerator {
         return policies.stream().filter(x -> Set.of("languages_policy").contains(x.policyName())).toList();
     }
 
+    public static String randomNonVector(List<Column> previousOutput) {
+        return randomName(previousOutput.stream().filter(x -> x.type().contains("vector") == false).toList());
+    }
+
     public static String randomName(List<Column> previousOutput) {
         String result = randomRawName(previousOutput);
         if (result == null) {

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/command/pipe/MvExpandGenerator.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/command/pipe/MvExpandGenerator.java
@@ -25,7 +25,7 @@ public class MvExpandGenerator implements CommandGenerator {
         List<EsqlQueryGenerator.Column> previousOutput,
         QuerySchema schema
     ) {
-        String toExpand = EsqlQueryGenerator.randomName(previousOutput);
+        String toExpand = EsqlQueryGenerator.randomNonVector(previousOutput);
         if (toExpand == null) {
             return EMPTY_DESCRIPTION; // no columns to expand
         }

